File: src/main/java/project/rpg/player/PlayerInformation.java
package project.rpg.player;

import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import project.rpg.Rpg;
import project.rpg.manager.AttributeManager;
import project.rpg.player.info.Skill;
import project.rpg.player.info.Status;
import project.rpg.ui.ActionBarUI;

import java.util.HashMap;
import java.util.Map;

import static project.rpg.player.info.Skill._skills;

public class PlayerInformation {

    private static final Map<Player, Integer> _mana = new HashMap<>();
    private final Rpg _plugin;
    public static void makeInfo(Player player){
        ActionBarUI.addPlayer(player);
        AttributeManager.setAttributes(player, new Status(player));

        player.setHealthScale(Status.getPlayerMap().get(player.getUniqueId()).getHealth()/100.0);
        player.setHealthScaled(true);
        player.setHealth(Status.getPlayerMap().get(player.getUniqueId()).getHealth()/100.0);

        _skills.put(player.getName(), new Skill());
    }

    public PlayerInformation(Rpg plugin){
        _plugin = plugin;
    }

    public static void useMana(Player player, int mana){
        _skills.get(player.getName()).useMana(mana);
    }

    public static int getMana(Player player){
        if (_skills.get(player.getName())!=null) {
            return _skills.get(player.getName()).getMana();
        } else {
            return 0;
        }
    }

    public static int getMaxMana(Player player) {
        if (_skills.get(player.getName())!=null) {
            return _skills.get(player.getName()).getMaxMana();
        } else {
            return 0;
        }
    }

    public static Map<String, Skill> getManaMap(){
        return _skills;
    }

    public void startManaRefilling(){
        Bukkit.getScheduler().scheduleSyncRepeatingTask(_plugin, ()->{
            for(Player player : Bukkit.getOnlinePlayers()){
                if(!_skills.containsKey(player.getName())){
                    _skills.put(player.getName(), new Skill(100));
                }
                else if(_skills.get(player.getName()).getMana()< _skills.get(player.getName()).getMaxMana()){
                    _skills.get(player.getName()).plusMana();
                }
            }
        },10, 10); //마나 회복이 지나치게 느림
    }
}

File: src/main/kotlin/project/rpg/player/PlayerInformation.kt
package project.rpg.player

import org.bukkit.entity.Player
import project.rpg.database.Database
import project.rpg.player.status.maxHealth
import project.rpg.player.status.base.StatusName
import project.rpg.ui.text.ActionBarUI

object PlayerInformation {
    fun makeInfo(player: Player) {
        val user = User.newUser(player)

        player.setAttributeMaxHealth(player.status.maxHealth / 10.0)
        player.healthScale = 100.0
        player.health = 1.0

        Database.readUser(player)
        user.status.applyChanges()

        ActionBarUI.addPlayer(player)
    }

    fun deleteInfo(player: Player) {
        Database.writeUser(player)
        ActionBarUI.deletePlayer(player)
        User.removeUser(player)
    }

    @JvmStatic
    fun updateHealth(player: Player) {
        player.setAttributeMaxHealth(player.status.getStatusValues(StatusName.HEALTH) / 10.0)
    }
}